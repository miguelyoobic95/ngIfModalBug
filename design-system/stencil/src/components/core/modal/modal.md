---
name: Modal
category: Core Components
---

Modals can be customized with multiple properties. Properties conform to the IModalEntry interface below:

```ts
export interface IModalEntry {
    heading?: string;
    headingIcon?: string;
    content?: HTMLElement;
    hasHeader?: boolean;
    hasFooter?: boolean;
    footerText?: string;
    primaryButtonText?: string;
    secondaryButtonText?: string;
    cssClass?: string;
    animationName?: string;
    animationProp?: any;
    primaryFn?: Function;
    withYooCtrl?: boolean;
    scrollEnabled?: boolean;
}
```

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`heading`|string|   |the heading of the modal, will only be displayed if `hasHeader` property is `true`|
|`headingIcon`|string| |the icon to display to the left of the heading|
|`content`|HTMLElement| |the content inside the body of the modal|
|`hasHeader`|boolean|`true`|defines if the modal has a header|
|`hasFooter`|boolean|`false`|defines if the modal has a footer|
|`footerText`|string||the text inside the footer|
|`primaryButtonText`|string|`Confirm`  |the text inside the primary button, will only be displayed if `hasFooter` property is `true`|
|`secondaryButtonText`|string|`Cancel`|the text inside the secondary button, will only be displayed if `hasFooter` property is `true`|
|`cssClass`|string| |space separated classes to add to the modal|
|`animationName`|string||adds an animation to the modal, see animation docs for options|
|`animationProp`|any|||
|`primaryFn`|Function||function triggered on the modal primary button click|
|`withYooCtrl`|boolean|specify whether the `yoo-modal-controller` is being used|
|`scrollEnabled`|boolean|false|specify whether the `yoo-slim-scroll` is being used|

### Basic

```yoo-modal-basic.html
    <div style="height: 500px; width: 500px;">
        <yoo-modal heading="Basic Modal" with-yoo-ctrl="true">
            <div >Test content</div>        
        </yoo-modal>
    </div>
```

### With Footer

```yoo-modal-footer.html
    <div style="height: 500px; width: 500px;">
        <yoo-modal class="accent" with-yoo-ctrl="true" headingIcon="yo-fire" heading="Modal with footer" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
            <div >Test content</div>
        </yoo-modal>
    </div>
```

### Without Header

```yoo-modal-no-header.html

    <yoo-modal has-footer="true" with-yoo-ctrl="true" has-header="false">
        <div >Test content</div>
    </yoo-modal>
```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-modal accent|
|Dark|`dark`|yoo-modal dark|
|Success|`success`|yoo-modal success|
|Danger|`danger`|yoo-modal danger|
|Info|`info`|yoo-modal info|
|Warning|`warning`|yoo-modal warning|
|Fullscreen|`fullscreen`|yoo-modal fullscreen|
|Drawer|`drawer`|yoo-modal drawer|

```yoo-modal-danger.html
    <yoo-modal class="danger" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal danger" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-success.html
    <yoo-modal class="success" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal success" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-info.html
    <yoo-modal class="info" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal info" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-warning.html
    <yoo-modal class="warning" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal warning" has-footer="true" primary-button-text="Confirm" secondary-button-text="Cancel">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-dark.html
    <yoo-modal class="dark" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal dark" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-drawer.html
    <yoo-modal class=" drawer accent" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Modal dark" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-fullscreen.html
    <yoo-modal class="success fullscreen" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Fullscreen Success" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

```yoo-modal-menu.html
    <yoo-modal class="accent menu" with-yoo-ctrl="true" heading-icon="yo-fire" heading="Menu Accent" has-footer="true" primary-button-text="Confirm" secondary-button-text="Dismiss">
        <div >Test content</div>
    </yoo-modal>
```

## Events

|Attr|Description|
|---|---|
|`modalPrimaryButtonClicked`|called when the primary button is clicked|
|`modalClosed`|called when the modal is closed|