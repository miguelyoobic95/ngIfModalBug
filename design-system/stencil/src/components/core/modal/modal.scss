@mixin class-color-modal($class, $color, $color-light) {
    :host(.#{$class}) {
        .outer-container {
            color: $text-color;
            .modal-header {
                background-color: $color-light;
                color: $color;
                border-bottom-color: $color;
            }
        }
        &:not(.fullscreen):not(.drawer):not(.menu) {
            .outer-container {
                border: 1px solid $color;
            }
        }
    }
}

@mixin scale-modal() {
    .outer-container {
        border-radius: 0rem;
        height: 100%;
        width: 100%;
        .modal-header {
            border-radius: 0rem;
            width: 100%;
            position: absolute;
        }
        .modal-body {
            padding-top: 5.0625rem;
        }
        .modal-footer {
            width: 100%;
            position: absolute;
            bottom: 0rem;
        }
    }
}

:host {
    &[padding] {
        .outer-container {
            .modal-body {
                padding: 1rem;
            }
        }
    }
    .outer-container {
        display: flex;
        flex-direction: column; //border: 1px solid cssvar(dark-10);
        background: cssvar(light);
        width: 100%;
        height: 100%;
        border-radius: 0.3125rem;
        .modal-header {
            //padding: 1.5625rem;
            //height: 3rem;
            padding: 1rem;
            @include border-top-radius(0.3125rem); //border-bottom: 1px solid cssvar(dark-20);
            box-shadow: $input-bar-shadow;
            align-items: center;
            z-index: 1;
            color: cssvar(black);
            .inner-header {
                @include text-truncate();
                padding: 0 1rem;
            }
            .modal-heading {
                font-size: 1rem;
                font-weight: 400;
            }
            .icon {
                padding: 0.5rem;
            }
            .close-icon {
                padding-right: 0.5rem;
                cursor: pointer;
            }
            .hide-icon {
                color: transparent;
            }
        }
        .modal-body {
            height: 100%;
            min-height: 10.9375rem;
            align-items: center;
            justify-content: center;
            .modal-scroll-container {
                height: 100%;
                width: 100%;
            }
            .modal-content {
                width: 100%;
            }
        }
        .modal-footer {
            //height: 4.0625rem;
            padding: 1rem; //border-top: 1px solid cssvar(dark-20);
            box-shadow: $input-bar-shadow;
            align-items: center;
            z-index: 1;
            .footer-buttons {
                flex: 1 1 33%;
                .primary-button,
                .secondary-button {
                    padding-bottom: 0.5rem;
                    padding-top: 0.5rem;
                    padding-right: 0.5rem;
                }
            }
        }
    }
}

:host(.custom-controller) {
    position: relative;
    z-index: 2000;
    .outer-container {
        height: auto;
    } // top: 50%;
    // left: 50%;
    // margin-left: -300px;
    // margin-top: -140px;
}

:host(.fullscreen) {
    position: absolute;
    top: 0rem;
    right: 0rem;
    bottom: 0;
    left: 0;
    margin-left: 0;
    margin-top: 0;
    @include scale-modal();
}

:host(.drawer) {
    position: absolute;
    top: 0rem;
    right: 0rem;
    bottom: 0;
    left: 66.66%;
    margin-left: 0;
    margin-top: 0;
    @include scale-modal();
    .outer-container {
        .modal-header {
            flex-direction: row-reverse;
        }
    }
}

:host(.menu) {
    top: 0rem;
    left: 0rem;
    height: 100%;
    width: 100%; //width: 33.3%;
    margin-left: 0;
    margin-top: 0;
    position: relative;
    @include scale-modal();
    .outer-container {
        .modal-header {
            position: relative;
            flex-direction: row;
        }
        .modal-body {
            padding-top: 0 !important;
        }
    }
}

:host(.action-sheet) {
    bottom: 0rem;
    left: 0rem;
    top: auto;
    width: 100%;
    margin-left: 0;
    margin-top: 0;
    @include scale-modal();
    .outer-container {
        border: none;
        background-color: transparent;
        .modal-body {
            justify-content: center;
            width: 100%;
        }
        /deep/ .modal-body>div {
            width: 100%;
            div[slot="[modal-content"] {
                width: 100%;
            }
        }
    }
}

:host(.language-selector) {
    .outer-container {
        min-height: 420px;
        border: 0 solid;
        box-shadow: 0 6px 19px 0 rgba(40, 47, 54, 0.08);
        width: 250px;
        color: cssvar(text-color);
        border-radius: 0.78125rem;
        .modal-body {
            background-color: cssvar(light);
            height: 100%;
        }
    }
}

:host(.simple) {
    .outer-container {
        .modal-header {
            background-color: cssvar(light) !important;
            border: none;
            .modal-heading {
                font-weight: normal;
            }
        }
    }
}

@include class-color-modal(accent, cssvar(accent), cssvar(accent-20));
@include class-color-modal(success, cssvar(success-110), cssvar(success-20));
@include class-color-modal(danger, cssvar(danger), cssvar(danger-20));
@include class-color-modal(info, cssvar(info), cssvar(info-20));
@include class-color-modal(warning, cssvar(warning), cssvar(warning-20));
@include class-color-modal(dark, cssvar(dark), cssvar(dark-20));