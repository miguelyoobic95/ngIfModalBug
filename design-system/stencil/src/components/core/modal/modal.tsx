import { Component, Method, Element, Prop, EventEmitter, Event, Listen } from '@stencil/core';
import { setAnimation } from '../../../utils/anim';
import { IAnimationProp } from '@shared/interfaces';
import { YooSlimScrollComponent } from '../../core/slim-scroll/slim-scroll';
import { resizeWindow } from '../../../utils/helpers';
@Component({
    tag: 'yoo-modal',
    styleUrl: 'modal.scss',
    scoped: true
})
export class YooModalComponent {

    @Prop() heading: string;
    @Prop() headingIcon: string;
    @Prop() content: HTMLElement;
    @Prop() primaryButtonText: string;
    @Prop() secondaryButtonText: string;
    @Prop() hasHeader: boolean = true;
    @Prop() hasFooter: boolean;
    @Prop() footerText: string;
    @Prop() cssClass: string;
    @Prop() animationName: string;
    @Prop() animationProp: IAnimationProp;
    @Prop() primaryFn: Function;
    @Prop() withYooCtrl: boolean;
    @Prop() scrollEnabled: boolean;

    @Event() primaryButtonClicked: EventEmitter<boolean>;
    @Event() closed: EventEmitter<boolean>;

    @Element() host: HTMLStencilElement;

    componentWillLoad() {

    }

    componentDidLoad() {
        this.setContentInModal();
        this.animation(true);
        resizeWindow(() => this.resize());
    }

    componentDidUpdate() {
        this.setContentInModal();
    }

    onPrimaryButtonClick(event): void {
        if (event) {
            this.primaryButtonClicked.emit(true);
            this.primaryFn();
        }
    }

    @Method()
    close(): void {
        if (this.animationProp) { this.animationProp.open = false; }
        this.closed.emit(true);
        this.animation(false);
    }

    @Method()
    resize() {
        let slim: YooSlimScrollComponent = this.host.querySelector('yoo-slim-scroll') as any;
        if (slim) {
            if (slim.height === this.getSizeContainer().height) {
                slim.refresh();
            } else {
                slim.height = this.getSizeContainer().height;
            }
        }
    }

    @Listen('rowNumberChanged')
    onInputBarRawChange() {
        // resize the modal to adjust slim scroll if the input bar take more or less space
        setTimeout(this.resize(), 100);
    }

    animation(open: boolean, sentFromClose?: boolean) {
        if (this.animationName) {
            if (this.animationName === 'sticky_up') {
                const padding = 16;
                const modalHeight = (this.host.querySelector('div.outer-container').clientHeight);
                setAnimation(this.animationName, this.host, { open: open, distance: (((window.innerHeight / 2) + padding) - (modalHeight / 2)), modalHeight: modalHeight });
            } else {
                setAnimation(this.animationName, this.host.querySelector('.outer-container') as HTMLElement, (sentFromClose ? { open: open, up: true } : (this.animationProp ? this.animationProp : { open: open })));
            }
        }
    }

    setContentInModal() {
        if (this.scrollEnabled) {
            if (this.content) {
                let slim = document.createElement('yoo-slim-scroll');
                this.host.querySelector('div.modal-scroll-container').appendChild(slim);
                slim.appendChild(this.content);
            }
            this.resize();
        } else if (this.content) {
            this.host.querySelector('div.modal-body').appendChild(this.content);
        }
    }

    getSizeContainer() {
        let modalContainer: HTMLElement = this.host.querySelector('.modal-scroll-container');
        let maxHeight = window.innerHeight;
        if (modalContainer) {
            let modalHeader: HTMLElement = this.host.querySelector('.modal-header');
            if (modalHeader) {
                maxHeight -= modalHeader.clientHeight;
            }
            let modalFooter: HTMLElement = this.host.querySelector('.modal-footer');
            if (modalFooter) {
                maxHeight -= modalFooter.clientHeight;
            }
            return { height: Math.min((modalContainer.clientHeight), maxHeight) + 'px', width: modalContainer.clientWidth + 'px' };
        }
        return { height: '', width: '' };
    }

    hostData() {
        return {
            class: {
                ['custom-controller']: this.withYooCtrl
            }
        };
    }

    render(): JSX.Element {
        return (
            <div class="outer-container">
                {this.hasHeader ?
                    <div class="modal-header" attr-layout="row" attr-layout-align="space-between">
                        <span class="hide-icon" ></span>
                        <div class="inner-header">
                            {this.headingIcon ? <span class="icon"><i class={this.headingIcon}></i></span> : null}
                            <span class="modal-heading">{this.heading}</span>
                        </div>
                        <span class="close-icon" onClick={() => this.close()} ><i class="yo-close"></i></span>
                    </div>
                    : null}
                <div class="modal-body">
                    <div class="modal-scroll-container">
                        <yoo-slim-scroll enabled={this.scrollEnabled}>
                            <slot />
                        </yoo-slim-scroll>
                    </div>
                </div>
                {this.hasFooter ?
                    <div class="modal-footer" attr-layout="row">
                        {this.footerText ? <div class="footer-text"><span>{this.footerText}</span></div> : null}
                        <div class="footer-buttons" attr-layout="row" attr-layout-align="end center">
                            {this.secondaryButtonText ? <div class="secondary-button squared"><yoo-button class="dark" onClick={() => this.close()} text={this.secondaryButtonText}></yoo-button></div> : null}
                            {this.primaryButtonText ? <div class="primary-button squared"><yoo-button class="accent" onClick={(event: UIEvent) => this.onPrimaryButtonClick(event)} text={this.primaryButtonText}></yoo-button></div> : null}
                            <slot name="footer-slot" />
                        </div>
                    </div> : null}
            </div>
        );
    }
}
