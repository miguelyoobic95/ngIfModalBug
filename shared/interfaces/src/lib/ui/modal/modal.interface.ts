import { IFormField } from '../../entities/form-field/form-field.interface';
import { ISlide } from '../../entities/slide/slide.interface';

export interface IModalEntry {
    heading?: string;
    headingIcon?: string;
    hasHeader?: boolean;
    hasFooter?: boolean;
    footerText?: string;
    content?: HTMLElement;
    primaryButtonText?: string;
    secondaryButtonText?: string;
    cssClass?: string;
    animationName?: string;
    animationProp?: any;
    primaryFn?: Function;
    withYooCtrl?: boolean;
    scrollEnabled?: boolean;
}

export interface IModalUpsertConfig {
    collectionName?: string | any;
    formDefinition?: Array<IFormField>;
    slides?: Array<ISlide>;
    suffix?: string;
    //extra options
    extraValidators?: Array<any>;
    extraButtons?: any;
    ignoreRequired?: boolean;
    readonly?: boolean;
    //text options
    title?: string;
    saveText?: string;
    cancelText?: string;
    allowEdit?: boolean;
    //visual options
    width?: string;
    height?: any;
    isFullscreen?: boolean;
    canMove?: boolean;
}