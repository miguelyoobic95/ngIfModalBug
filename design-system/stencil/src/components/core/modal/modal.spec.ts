import { TestWindow } from '@stencil/core/testing';

jest.mock('../../../utils/anim');

import { YooModalComponent } from './modal';

fdescribe('YooModalComponent', () => {
    it('should build', () => {
        expect(new YooModalComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it('Should have a heading and a header', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal heading="this is text"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it('Should have no header', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal has-header="false"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it('Should have a heading icon', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal heading="this is text" heading-icon="yo-fire"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it('Should have a footer', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal has-footer="true"></yoo-modal>'
            });
            expect(element).toMatchSnapshot();
        });

        it('Should have content', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: `<yoo-modal>
                        <div >Test content</div>
                        </yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it('Should have custom primary button text', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: `<yoo-modal primary-button-text="Hello"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it('Should have custom secondary button text', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: `<yoo-modal secondary-button-text="Hello"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it('Should be a full screen modal', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: `<yoo-modal css-class="fullscreen"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });

        it('Should be a drawer modal', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: `<yoo-modal css-class="drawer"></yoo-modal>`
            });
            expect(element).toMatchSnapshot();
        });
    });

    describe('rendering', () => {
        it('Should render no heading if hasHeader equals false', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal has-header="false" heading="I should not exist"></yoo-modal>'
            });
            expect(element.getElementsByClassName('modal-header')).toHaveLength(0);
            expect(element.getElementsByClassName('modal-heading')).toHaveLength(0);
        });

        it('Should render no footer by default', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal></yoo-modal>'
            });
            expect(element.getElementsByClassName('modal-footer')).toHaveLength(0);
        });

        // TO DO: fix modal entry tests
        describe('modal-entry-rendering', () => {
            it('Should render content', async () => {
                let window = new TestWindow();
                let element = await window.load({
                    components: [YooModalComponent],
                    html: '<yoo-modal></yoo-modal>'
                });
                let htmlElement = document.createElement('div');
                htmlElement.innerHTML = `<div class="test">This is the test text</div>`;
                let modalProps = { content: htmlElement };
                element = await Object.assign(element, modalProps);
                await window.flush();
                //expect(element.querySelector('div.test')).toBeTruthy();
                //expect(element.querySelector('div.test').innerHTML).toEqual('This is the test text');
            });

            it('Should render heading', async () => {
                let window = new TestWindow();
                let element = await window.load({
                    components: [YooModalComponent],
                    html: '<yoo-modal></yoo-modal>'
                });
                let modalProps = { heading: 'Arigato' };
                element = await Object.assign(element, modalProps);
                expect(element.heading).toEqual('Arigato');
                await window.flush();
                // expect(element.getElementsByClassName('modal-heading')).toHaveLength(1);
                // expect(element.getElementsByClassName('modal-heading')[0].innerHTML).toEqual('Arigato');
            });
        });
    });

    describe('events', () => {
        let window, element, res, emitEvent;
        beforeEach(async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooModalComponent],
                html: '<yoo-modal has-footer="true" primary-button-text="primary" secondary-button-text="secondary"></yoo-modal>'
            });
            emitEvent = jest.fn(ev => { res = ev.detail; });
        });
        it('should emit modalPrimaryButtonClicked on primary button click', async () => {
            window.document.addEventListener('primaryButtonClicked', emitEvent);
            let button = element.querySelector('.primary-button > yoo-button');
            await button.click();
            expect(emitEvent).toHaveBeenCalled();
            expect(res).toEqual(true);
        });
        it('should emit modalClosed on secondary button click', async () => {
            window.document.addEventListener('closed', emitEvent);
            let button = element.querySelector('.secondary-button > yoo-button');
            await button.click();
            expect(emitEvent).toHaveBeenCalled();
            expect(res).toEqual(true);
        });
    });
});
