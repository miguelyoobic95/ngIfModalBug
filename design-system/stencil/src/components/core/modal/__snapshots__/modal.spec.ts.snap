// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooModalComponent Snapshots Should be a drawer modal 1`] = `
<yoo-modal
  class="hydrated"
  css-class="drawer"
  data-yoo-modal-host=""
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should be a full screen modal 1`] = `
<yoo-modal
  class="hydrated"
  css-class="fullscreen"
  data-yoo-modal-host=""
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have a footer 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  has-footer="true"
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    <div
      attr-layout="row"
      class="modal-footer"
      data-yoo-modal=""
    >
      
      <div
        attr-layout="row"
        attr-layout-align="end center"
        class="footer-buttons"
        data-yoo-modal=""
        data-yoo-modal-slot=""
      >
        
        
      </div>
    </div>
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have a heading and a header 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  heading="this is text"
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          this is text
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have a heading icon 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  heading="this is text"
  heading-icon="yo-fire"
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        <span
          class="icon"
          data-yoo-modal=""
        >
          <i
            class="yo-fire"
            data-yoo-modal=""
          />
        </span>
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          this is text
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have content 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
>
  
  
  
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
          
                        
          <div>
            Test content
          </div>
          
                        
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have custom primary button text 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  primary-button-text="Hello"
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have custom secondary button text 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  secondary-button-text="Hello"
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="modal-header"
      data-yoo-modal=""
    >
      <span
        class="hide-icon"
        data-yoo-modal=""
      />
      <div
        class="inner-header"
        data-yoo-modal=""
      >
        
        <span
          class="modal-heading"
          data-yoo-modal=""
        >
          
        </span>
      </div>
      <span
        class="close-icon"
        data-yoo-modal=""
      >
        <i
          class="yo-close"
          data-yoo-modal=""
        />
      </span>
    </div>
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;

exports[`YooModalComponent Snapshots Should have no header 1`] = `
<yoo-modal
  class="hydrated"
  data-yoo-modal-host=""
  has-header="false"
>
  
  <div
    class="outer-container"
    data-yoo-modal=""
  >
    
    <div
      class="modal-body"
      data-yoo-modal=""
    >
      <div
        class="modal-scroll-container"
        data-yoo-modal=""
      >
        <yoo-slim-scroll
          data-yoo-modal=""
          data-yoo-modal-slot=""
        >
          
        </yoo-slim-scroll>
      </div>
    </div>
    
  </div>
</yoo-modal>
`;
